Lib/idlelib/pyshell.py

```diff
@@ -902,10 +902,11 @@ def __init__(self, flist=None):
         self.save_stderr = sys.stderr
         self.save_stdin = sys.stdin
         from idlelib import iomenu
-        self.stdin = PseudoInputFile(self, "stdin", iomenu.encoding)
-        self.stdout = PseudoOutputFile(self, "stdout", iomenu.encoding)
-        self.stderr = PseudoOutputFile(self, "stderr", iomenu.encoding)
-        self.console = PseudoOutputFile(self, "console", iomenu.encoding)
+        self.stdin = StdInFile(self, "stdin", iomenu.encoding)
```

Comment:
This code is the same in 2.7.  I have been thinking that for 3.x, we should replace 'iomenu.encoding' with 'utf-8' at least here if not elsewhere.  I believe it will nearly(?) always be 'utf-8' but if it can be different, it seems a mistake to use anything else to transport 3.x strings between run and idle processes.